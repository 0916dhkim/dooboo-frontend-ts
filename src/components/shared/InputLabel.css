@import '../../root.css';

.root {
  padding-top: 28px;
  width: 100%;
  display: flex;
  flex-direction: column-reverse;
}

.input {
  padding-right: 12px;
  height: 44px;
  font-size: 18px;
  line-height: 27px;
  border-top: none;
  border-left: none;
  border-right: none;
  &:focus {
    outline: none;
    border-image-source: linear-gradient(to right, rgb(12, 157, 197), rgb(201, 109, 216));
    border-image-slice: 1;
  }
}

.label {
  font-size: 14px;
  line-height: 20px;
  text-align: left;
  font-weight: normal;
}

.labelBefore {
  font-size: 14px;
  line-height: 20px;
  text-align: left;
  font-weight: normal;
  color: #9b9b9b;
}

.labelAfter {
  font-size: 14px;
  line-height: 20px;
  text-align: left;
  font-weight: bold;
  color: #4a4a4a;
}
