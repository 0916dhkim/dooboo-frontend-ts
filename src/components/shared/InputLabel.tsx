import React , { Component } from 'react';

const styles = require('./InputLabel.css');

interface IProps {
  containerStyle: object;
  type: string;
  placeholder: string;
  labelTxt: string;
  onTxtChanged: any;
}

interface IState {
  inputDone: boolean;
  isFocus: boolean;
}

export class InputLabel extends Component<IProps, IState> {
  private inputValue: HTMLInputElement;

  constructor(props) {
    super(props);
    this.state = {
      isFocus: false,
      inputDone: false,
    };
  }

  public onTxtChanged(e) {
    const v: string = e.target.value.toString();
    if (v.length > 0) {
      this.setState({ inputDone: true });
    } else {
      this.setState({ inputDone: false});
    }

    this.props.onTxtChanged(v);
  }

  public render() {
    return (
      <div
        className={styles.root}
        style={this.props.containerStyle}
      >
        <label
          className={
            this.state.inputDone
              ? styles.labelAfter
              : this.state.isFocus
                ? styles.label
                : styles.labelBefore
          }
        >{this.props.labelTxt}</label>
        <input
          ref={(v) => this.inputValue = v}
          className={styles.input}
          type={this.props.type}
          onFocus={ () => this.onFocus(true) }
          onBlur={ () => this.onFocus(false) }
          style={
            this.state.inputDone
              ? { borderBottom: '2px solid #4a4a4a' }
              : { borderBottom: '2px solid #9b9b9b' }
          }
          placeholder={this.props.placeholder}
          onChange={(e) => this.onTxtChanged(e)}
        />
      </div>
    );
  }

  private onFocus = (state: boolean) => {
    this.setState({ isFocus: state });
  }
}

export default InputLabel;
