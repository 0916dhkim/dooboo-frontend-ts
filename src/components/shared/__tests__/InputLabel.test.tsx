import React from 'react';
import { shallow } from 'enzyme';

import { InputLabel } from '@shared/InputLabel';

describe('InputLabel component test', () => {
  const component = shallow(
    <InputLabel
      type='email'
      placeholder='email'
      labelTxt='email'
      containerStyle={{
        margin: '0 20px',
      }}
      onTxtChanged={ () => { return; }}
    />,
  );

  it('component matches with its snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('the component input event is triggered with a mock argument and the state will be changed', () => {
    const input = component.find('input');
    input.simulate('change', { target: {id: 'emailInput', value: 'mock-email' }});
    expect(component.state().inputDone).toEqual(true);
  });
});
