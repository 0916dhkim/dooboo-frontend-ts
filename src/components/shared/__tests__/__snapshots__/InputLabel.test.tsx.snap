// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel component test component matches with its snapshot 1`] = `
<div
  className="root"
  style={
    Object {
      "margin": "0 20px",
    }
  }
>
  <label
    className="labelBefore"
  >
    email
  </label>
  <input
    className="input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="email"
    style={
      Object {
        "borderBottom": "2px solid #9b9b9b",
      }
    }
    type="email"
  />
</div>
`;
